gms/gossiper.cc

```diff
@@ -643,26 +644,31 @@ future<> gossiper::apply_state_locally(std::map<inet_address, endpoint_state> ma
         std::chrono::steady_clock::now() - start).count());
 }
 
-future<> gossiper::force_remove_endpoint(inet_address endpoint) {
+future<> gossiper::force_remove_endpoint(inet_address endpoint, permit_id pid) {
     if (endpoint == get_broadcast_address()) {
         return make_exception_future<>(std::runtime_error(format("Can not force remove node {} itself", endpoint)));
     }
-    return container().invoke_on(0, [endpoint] (auto& gossiper) mutable -> future<> {
+    return container().invoke_on(0, [endpoint, pid] (auto& gossiper) mutable -> future<> {
+        auto permit = co_await gossiper.lock_endpoint(endpoint, pid);
+        pid = permit.id();
```

Comment:
This is weird. A permit_id is provided. Then we reassign it with a the new one.

The force_remove_endpoint is always called with a null id. Why not drop the pid parameter to avoid confusing.

Reply:
> This is weird. A permit_id is provided. Then we reassign it with a the new one.

If the passed permit_id is null then the permit lock_endpoint returns will have a random one.

> The force_remove_endpoint is always called with a null id. Why not drop the pid parameter to avoid confusing.

I plan to add a call to force_remove_endpoint in https://github.com/scylladb/scylladb/pull/14471
that may pass a valid permit_id.

Reply:
I know it is null id then a random id which is generated.